R/pkg/R/generics.R

```diff
@@ -396,67 +397,81 @@ setGeneric("agg", function (x, ...) { standardGeneric("agg") })
 #' @param object x a SparkDataFrame or a Column
 #' @param data new name to use
 #' @return a SparkDataFrame or a Column
+#' @noRd
 NULL
 
 #' @rdname arrange
 #' @export
+#' @noRd
 setGeneric("arrange", function(x, col, ...) { standardGeneric("arrange") })
 
 #' @rdname as.data.frame
 #' @export
+#' @noRd
 setGeneric("as.data.frame",
            function(x, row.names = NULL, optional = FALSE, ...) {
              standardGeneric("as.data.frame")
            })
 
 #' @rdname attach
 #' @export
+#' @noRd
 setGeneric("attach")
 
 #' @rdname cache
 #' @export
+#' @noRd
 setGeneric("cache", function(x) { standardGeneric("cache") })
 
 #' @rdname checkpoint
 #' @export
+#' @noRd
 setGeneric("checkpoint", function(x, eager = TRUE) { standardGeneric("checkpoint") })
 
 #' @rdname coalesce
 #' @param x a Column or a SparkDataFrame.
```

Comment:
Thanks for pointing this out. In this case, it should be `a Column` not `a SparkDataFrame or a Column`.  Indeed, this is a problem of documenting only the generics and inheriting the parameter doc from the generics in each method. Ideally, each specific method of a generic should have its own documentation, not the way we document the generics now.  Suppose we have a new method `alias` for another class, the doc in the generics needs to be updated to include this new class. But it's not clear to the developer where to make the change. If we instead just document each method of the generics, it will be much clearer. 
I will do a thorough cleanup of these issues after we decide on the big items.